pallets/pool-xyk/src/migrations/v3.rs

```diff
@@ -0,0 +1,207 @@
+// This file is part of the SORA network and Polkaswap app.
+
+// Copyright (c) 2020, 2021, Polka Biome Ltd. All rights reserved.
+// SPDX-License-Identifier: BSD-4-Clause
+
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+
+// Redistributions of source code must retain the above copyright notice, this list
+// of conditions and the following disclaimer.
+// Redistributions in binary form must reproduce the above copyright notice, this
+// list of conditions and the following disclaimer in the documentation and/or other
+// materials provided with the distribution.
+//
+// All advertising materials mentioning features or use of this software must display
+// the following acknowledgement: This product includes software developed by Polka Biome
+// Ltd., SORA, and Polkaswap.
+//
+// Neither the name of the Polka Biome Ltd. nor the names of its contributors may be used
+// to endorse or promote products derived from this software without specific prior written permission.
+
+// THIS SOFTWARE IS PROVIDED BY Polka Biome Ltd. AS IS AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL Polka Biome Ltd. BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::pallet::{Config, Pallet};
+use common::{LiquiditySourceType, ToFeeAccount, TradingPair, XSTUSD};
+use frame_support::pallet_prelude::{Get, StorageVersion};
+use frame_support::traits::OnRuntimeUpgrade;
+use frame_support::{log::info, weights::Weight};
+use sp_std::prelude::Vec;
+
+use crate::{AccountPools, PoolProviders, Properties, Reserves, TotalIssuances, WeightInfo};
+use ceres_liquidity_locker::LockerData;
+use demeter_farming_platform::UserInfos;
+use trading_pair::EnabledSources;
+
+pub struct XYKPoolUpgrade<T, L>(core::marker::PhantomData<(T, L)>);
+
+/// Migration which removes invalid pools from `XYKPool` and their corresponding dependencies.
+impl<T, L> OnRuntimeUpgrade for XYKPoolUpgrade<T, L>
+where
+    T: crate::Config,
+    <T as frame_system::Config>::AccountId: From<[u8; 32]>,
+    L: Get<Vec<(T::AssetId, T::AssetId, T::DEXId)>>,
+{
+    fn on_runtime_upgrade() -> Weight {
```

Comment:
It would be nice to decompose this function into smaller functions to make it more readable